### PR TITLE
ODC-7680: Migrate `FilterTable` to PF5

### PR DESCRIPTION
**Pull request by logonoff**
**Fixes**: 

https://issues.redhat.com/browse/ODC-7680

**Solution Description**: 

migrate file to PF5 tables and remove unlocalized strings
 
**Screen shots / Gifs for design review**: 

https://github.com/user-attachments/assets/b1c369de-e42e-4e54-b760-718d05280b48

**Unit test coverage report**: 

n/a

**Test setup:**

install knative operator, set up knative servicing, then add a trigger with a filter

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/overview/FilterTable.tsx [hunk 1]
```diff
@@ -39,21 +36,44 @@ const FilterTable: React.FC<FilterTableProps> = ({
       ],
     };
   };
+
+  const { t } = useTranslation('knative-plugin');
+
   const data = {
-    columns: ['Key', 'Value'],
+    columns: [t('Key'), t('Value')],
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/FilterTable.tsx:43]: Change `Key` to `Attribute` to make it consistent with Trigger form label for filter

```suggestion
    columns: [t('Attribute'), t('Value')],
```

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Comment by logonoff** [comment]
adding labels because this is debt

/label docs-approved
/label px-approved

**Comment by sanketpathak** [comment]
Verified on cluster with admin and non-admin user
/label qe-approved

**Pull request merged**